### ISSUE TITLE
Include getopt.h in tsql.c

### ISSUE THREAD

**Issue by l8nite** [FreeTDS/freetds#45]
I was trying to get the latest to install on alpine linux, stole this from http://git.alpinelinux.org/cgit/aports/tree/main/freetds/fix-includes.patch

See this PR: https://github.com/rails-sqlserver/tiny_tds/pull/257

**Comment by freddy77** [comment]
Looks like it would be much better to include it in include/replacements.h if HAVE_GETOPT_H is defined.
Can you do this test? It works for me.

**Comment by freddy77** [comment]
diff --git a/include/replacements.h b/include/replacements.h
index 7e29587..b55c980 100644
--- a/include/replacements.h
+++ b/include/replacements.h
@@ -33,6 +33,9 @@
 #ifdef HAVE_LIBGEN_H
 # include <libgen.h>
 #endif
+#ifdef HAVE_GETOPT_H
+# include <getopt.h>
+#endif
 
 #if !HAVE_POLL
 #include <replacements/poll.h>

**Comment by l8nite** [comment]
Hm, that still failed to compile with the error message I received pre-patch:

```
tsql.c: In function 'reset_getopt':
tsql.c:331:2: error: 'optreset' undeclared (first use in this function)
  optreset = 1;
```

**Comment by freddy77** [comment]
I bet HAVE_GETOPT_H is not defined in include/config.h.
Try to just apply



diff --git a/include/replacements.h b/include/replacements.h
index 7e29587..b55c980 100644
--- a/include/replacements.h
+++ b/include/replacements.h
@@ -33,6 +33,7 @@
 \#ifdef HAVE_LIBGEN_H
 \#include \<libgen.h>
 \#endif
+\#include \<getopt.h>

 \#if !HAVE_POLL
 \#include

**Comment by l8nite** [comment]
That did the trick. :+1: Want me to submit a PR?

**Comment by freddy77** [comment]
this issue is enough

**Comment by freddy77** [comment]
Fixed both 0.95 and master!
